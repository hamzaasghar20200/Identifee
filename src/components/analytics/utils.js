import { floorFigure, parsePercentage, setDateFormat } from '../../utils/Utils';

export const formatQueryResult = (field, result, annotation, opts = {}) => {
  const { measures, dimensions } = annotation;

  if (!measures[field] && !dimensions[field]) {
    return result[field];
  }

  const attrs = measures[field] || dimensions[field];

  if (!attrs) {
    return result[field];
  }

  if (attrs.format === 'currency') {
    return floorFigure(result[field]);
  }
  if (attrs.format === 'percent') {
    return parsePercentage(result[field]);
  }
  if (attrs.type === 'time') {
    return setDateFormat(result[field], opts.format);
  }
  if (
    attrs.type === 'number' &&
    attrs.meta &&
    attrs.meta.type === 'date-diff'
  ) {
    return result[field].days || 0;
  }
  return result[field];
};
