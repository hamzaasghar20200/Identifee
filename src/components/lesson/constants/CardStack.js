export const stacks = [
  {
    id: 1,
    top: -5,
    left: 0,
  },
  {
    id: 2,
    top: -10,
    left: 0,
  },
  {
    id: 3,
    top: -15,
    left: 0,
  },
];
