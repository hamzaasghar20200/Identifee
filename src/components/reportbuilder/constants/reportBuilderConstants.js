import React from 'react';
import TextBlock from '../blocks/TextBlock';
import BlockDescription from '../blocks/BlockDescription';
import PartnerLogoBlock from '../blocks/PartnerLogoBlock';
import IconTextBlock from '../blocks/IconTextBlock';
import CircleChartTextBlock from '../blocks/CircleChartTextBlock';
import CalendarBlock from '../blocks/CalendarBlock';
import RPMGBlock from '../blocks/RPMGBlock';
import RawHTMLBlock from '../blocks/RawHTMLBlock';

export const ComponentTextTypes = {
  Donut: 'donut',
  Calendar: 'calendar',
  PercentText: 'percentText',
  IconText: 'iconText',
  DonutSelection: 'donutSelection',
  Static: 'static',
};

export const ComponentSourceTypes = {
  RPMG: 'rpmg',
  SPGlobal: 'spGlobal',
  FasterPayments: 'fasterPayments',
  Custom: 'custom',
};

export const PartnerLogoBySource = {
  rpmg: '/img/integrations/RPMG-Logo.png',
  spGlobal: '/img/integrations/sp-global.png',
  fasterPayments: '/img/integrations/FasterPayments-Logo.jpeg',
  custom: '',
};

export const ReportBlockControls = [
  {
    id: 3,
    icon: 'add_circle',
    type: 'addNew',
    tooltip: 'Insert a new block',
    isLoading: false,
  },
  {
    id: 5,
    icon: 'swap_horiz',
    type: 'changePosition',
    tooltip: 'Revert',
    isLoading: false,
  },
  {
    id: 6,
    icon: 'block',
    type: 'remove',
    tooltip: 'Disable',
    isLoading: false,
  },
];

export const ReportBlocksType = [
  { id: 1, key: 'All', name: 'All Blocks' },
  { id: 2, key: ComponentSourceTypes.SPGlobal, name: 'S&P Global' },
  { id: 3, key: ComponentSourceTypes.RPMG, name: 'RPMG' },
  { id: 4, key: ComponentSourceTypes.FasterPayments, name: 'Faster Payments' },
  { id: 5, key: ComponentSourceTypes.Custom, name: 'Custom' },
];

export const defaultReportBlocks = [
  {
    id: 1,
    direction: 'flex-row-reverse',
    description: [
      'The average Days Sales Outstanding (DSO) of your peer group.',
    ],
    data: {
      value: '64',
      text: 'Days',
    },
    type: 'Calendar',
    partnerLogo: {
      src: '/img/integrations/sp-global.png',
      position: 'float-right',
      style: { width: 100 },
    },
  },
  {
    id: 2,
    direction: '',
    description: [
      'On average, your peers pay (36%) of all payables between $2,500 - $10,000 by commercial card, providing a substantial increase in Days Payables.',
      'Peers are leveraging commercial card to improve Working Capital.',
    ],
    data: {
      labels: ['Card', 'Others'],
      data: {
        data: [36, 64],
        backgroundColor: ['#5cb85c', '#eee'],
      },
    },
    type: 'RPMG',
    partnerLogo: {
      src: '/img/integrations/RPMG-Logo.png',
      position: 'float-left',
    },
    options: { cutout: 40 },
  },
  {
    id: 3,
    direction: 'flex-row-reverse',
    description: [
      'A savings of $2.50 per item can be realized by moving from checks to digital payments.',
    ],
    data: {
      value: '$2.50',
      text: 'Savings',
    },
    type: 'Text',
    partnerLogo: {
      src: '/img/integrations/FasterPayments-Logo.jpeg',
      position: 'float-right',
    },
  },
  {
    id: 4,
    direction: '',
    description: [
      'The average Days Sales Outstanding (DSO) of your peer group.',
    ],
    data: {
      value: '45',
      text: 'Days',
    },
    type: 'Calendar',
    partnerLogo: {
      src: '/img/integrations/sp-global.png',
      position: 'float-left',
      style: { width: 100 },
    },
  },
  {
    id: 5,
    direction: 'flex-row-reverse',
    description: [
      'On average, your peers pay (40%) of all payables between $2,500 - $10,000 by ACH.',
    ],
    data: {
      labels: ['ACH', 'Others'],
      data: {
        data: [40, 60],
        backgroundColor: ['#c1232a', '#eee'],
      },
    },
    type: 'CircleChartText',
    partnerLogo: {
      src: '/img/integrations/RPMG-Logo.png',
      position: 'float-right',
    },
    options: { cutout: 40 },
  },
  {
    id: 6,
    direction: '',
    description: [
      'Faster payments networks now enable instant payments to be sent to approximately 80% of the DDAs in the United States.',
    ],
    data: {
      value: '80%',
      text: 'DDAs',
    },
    type: 'Text',
    partnerLogo: {
      src: '/img/integrations/FasterPayments-Logo.jpeg',
      position: 'float-left',
    },
  },
  {
    id: 7,
    direction: 'flex-row-reverse',
    iconConfig: { icon: 'dvr', color: 'text-blue' },
    description: [
      'Faster payments networks now enable instant payments to be sent to approximately 80% of the DDAs in the United States.',
    ],
    data: {
      text: 'Faster Payments',
    },
    type: 'IconText',
    partnerLogo: {
      src: '/img/integrations/FasterPayments-Logo.jpeg',
      position: 'float-right',
    },
  },
  {
    id: 8,
    direction: 'flex-row-reverse',
    iconConfig: { icon: 'info', color: 'text-black' },
    description: [
      '*Does not display all payment types. For further details, please review your Account Analysis statement.',
      '*Bank Deposit Assessment fess are fess associated with FDIC.',
    ],
    data: {
      text: 'Legal',
    },
    type: 'IconText',
    partnerLogo: {},
  },
];

export const DynamicBlock = ({ type, children, ...props }) => {
  switch (type) {
    case ComponentTextTypes.Static:
      return React.cloneElement(
        <RawHTMLBlock
          text={props.text}
          {...props}
          partner={
            <PartnerLogoBlock
              logo={props.partnerLogo.src}
              placement={props.partnerLogo.position}
              style={props.partnerLogo.style}
            />
          }
        />
      );
    case 'RPMG':
    case ComponentTextTypes.DonutSelection:
    case ComponentTextTypes.Donut:
      return React.cloneElement(
        <RPMGBlock
          {...props}
          text={<BlockDescription texts={props.description} />}
          partner={
            <PartnerLogoBlock
              logo={props.partnerLogo.src}
              placement={props.partnerLogo.position}
              style={props.partnerLogo.style}
            />
          }
        />
      );
    case 'Text':
      return React.cloneElement(
        <TextBlock
          {...props}
          text={<BlockDescription texts={props.description} />}
          partner={
            <PartnerLogoBlock
              logo={props.partnerLogo.src}
              placement={props.partnerLogo.position}
              style={props.partnerLogo.style}
            />
          }
        />
      );
    case 'IconText':
    case ComponentTextTypes.IconText:
      return React.cloneElement(
        <IconTextBlock
          {...props}
          text={<BlockDescription texts={props.description} />}
          partner={
            <PartnerLogoBlock
              logo={props.partnerLogo.src}
              placement={props.partnerLogo.position}
              style={props.partnerLogo.style}
            />
          }
        />
      );
    case 'CircleChartText':
      return React.cloneElement(
        <CircleChartTextBlock
          {...props}
          text={<BlockDescription texts={props.description} />}
          partner={
            <PartnerLogoBlock
              logo={props.partnerLogo.src}
              placement={props.partnerLogo.position}
              style={props.partnerLogo.style}
            />
          }
        />
      );
    case 'Calendar':
    case ComponentTextTypes.Calendar:
      return React.cloneElement(
        <CalendarBlock
          {...props}
          text={<BlockDescription texts={props.description} />}
          partner={
            <PartnerLogoBlock
              logo={props.partnerLogo.src}
              placement={props.partnerLogo.position}
              style={props.partnerLogo.style}
            />
          }
        />
      );
  }
};
