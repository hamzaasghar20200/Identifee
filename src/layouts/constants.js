export const getClientPortalOrganization = () => {
  return JSON.parse(sessionStorage.getItem('organizationObj'));
};

export const getClientPortalToken = () => {
  return JSON.parse(sessionStorage.getItem('tokenPayload'));
};
