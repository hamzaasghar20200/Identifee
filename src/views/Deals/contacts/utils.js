import userService from '../../../services/user.service';
import {
  ADDRESS_STATE,
  ADDRESS_STREET,
  DEAL_TYPE,
  EMAIL_LOCATION,
  EMAIL_LOCATION_1,
  EMAIL_LOCATION_2,
  PHONE_LOCATION,
  PHONE_LOCATION_2,
  PHONE_LOCATION_3,
  PHONE_LOCATION_4,
  STATUS,
} from '../../../utils/constants';
import {
  initialFilters,
  initialOrgForm,
  initialPeopleForm,
  initialDealsForm,
} from './Contacts.constants';

export const onGetOwners = (
  filter,
  dataCallback,
  existingOwners,
  extraData
) => {
  async function onGetUsers() {
    const response = await userService
      .getUsers(
        { ...filter, status: 'active' },
        {
          limit: 10,
          extraData,
        }
      )
      .catch((err) => err);

    const { data } = response || {};
    let users = [];

    const addName = (u) => ({ ...u, name: `${u.first_name} ${u.last_name}` });

    if (existingOwners && Object.keys(data).length) {
      // only interested in users who are not existing owners
      const existingOwnerIds = existingOwners.map(({ user_id }) => user_id);
      users = data.users
        .filter(({ id }) => !existingOwnerIds.includes(id))
        .map(addName);
    }

    return dataCallback(
      users.length
        ? users.filter((i) => i?.first_name && i?.last_name).map(addName)
        : !existingOwners
        ? data?.users.filter((i) => i?.first_name && i?.last_name).map(addName)
        : []
    );
  }

  onGetUsers();
};

export const onInputSearch = async (e, data, setData) => {
  if (!e) {
    return setData({
      search: '',
    });
  }

  const { value } = e.target;

  setData({
    ...data,
    search: value || '',
  });
};

export const onHandleSelect = (item, name, dispatch, setSelect) => {
  if (name === STATUS || name === DEAL_TYPE) setSelect(item.title);

  if (name === 'lead_source') setSelect(item.name);

  if (name === 'label_id') {
    return dispatch({
      type: 'set',
      input: name,
      payload: item.target.value,
    });
  }

  if (name === DEAL_TYPE) {
    return dispatch({
      type: 'set',
      input: name,
      payload: item.id,
    });
  }

  if (name === 'organization_id' && item.organization_id) {
    return dispatch({
      type: 'set',
      input: name,
      payload: item.organization_id,
    });
  }

  const payloadFields = [
    EMAIL_LOCATION,
    EMAIL_LOCATION_1,
    EMAIL_LOCATION_2,
    PHONE_LOCATION,
    PHONE_LOCATION_2,
    PHONE_LOCATION_3,
    PHONE_LOCATION_4,
    ADDRESS_STATE,
    ADDRESS_STREET,
  ];

  dispatch({
    type: 'set',
    input: name,
    payload:
      payloadFields.includes(name) ||
      name.includes(PHONE_LOCATION) ||
      name.includes(EMAIL_LOCATION)
        ? item.name
        : item.id,
  });
};

export const onInputChange = (e, dispatch) => {
  const { name, value, type, checked } = e.target;

  if (name === 'amount') {
    const preg = /^([0-9]+\.?[0-9]{0,2})$/;

    if (preg.test(value)) {
      return dispatch({
        type: 'set',
        input: name,
        payload: value,
      });
    } else if (value < 2) {
      dispatch({
        type: 'set',
        input: name,
        payload: value,
      });
    }
    return;
  }
  dispatch({
    type: 'set',
    input: name,
    payload:
      type === 'number' ? Number(value) : type === 'checkbox' ? checked : value,
  });
};

export const onInputChangeAmount = (e, dispatch, data) => {
  const { name, value, type, checked } = e.target;

  if (name === 'amount') {
    const preg = /^([0-9]+\.?[0-9]{0,2})$/;

    if (preg.test(value)) {
      return dispatch({
        ...data,
        type: 'set',
        input: name,
        amount: value,
      });
    } else if (value < 2) {
      dispatch({
        ...data,
        type: 'set',
        input: name,
        amount: value,
      });
    }
    return;
  }
  dispatch({
    ...data,
    type: 'set',
    input: name,
    amount:
      type === 'number' ? Number(value) : type === 'checkbox' ? checked : value,
  });
};

export const reducer = (state, action) => {
  switch (action.type) {
    case 'load':
      return {
        ...action.payload,
      };
    case 'add_email':
    case 'add_phone':
      return {
        ...state,
        ...action.payload,
      };
    case 'set':
      return {
        ...state,
        [action.input]: action.payload,
      };
    case 'remove':
      delete state[action.input];

      return {
        ...state,
      };
    case 'reset':
      return initialFilters;
    case 'reset-peopleForm':
      return initialPeopleForm;
    case 'reset-orgForm':
      return initialOrgForm;
    case 'reset-dealForm':
      return initialDealsForm;
    default:
      return state;
  }
};

export const tableFiltersReducer = (state, action) => {
  switch (action.type) {
    case 'set':
      return {
        ...state,
        [action.input]: action.fullPayLoad,
      };
    case 'remove':
      delete state[action.input];

      return {
        ...state,
      };
    case 'reset':
      return {};
    default:
      return state;
  }
};

export const changePaginationPage = (newPage, setPaginationPage) => {
  setPaginationPage((prev) => ({ ...prev, page: newPage }));
};

export const onHandleCloseModal = (dispatch, toggle, type) => {
  dispatch({
    type,
  });

  toggle();
};

export const removeCustomFieldsFromActivityForm = (
  formValue,
  customFieldData
) => {
  const tempContactForm = { ...formValue };

  customFieldData.forEach((customField) => {
    const keyToRemove = customField.value;
    Object.entries(tempContactForm).forEach(([key, value]) => {
      if (value === keyToRemove) {
        delete tempContactForm[key];
      }
    });
  });
  return tempContactForm;
};
