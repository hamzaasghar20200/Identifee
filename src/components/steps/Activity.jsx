import React, { useState, useEffect } from 'react';
import { Col, Row } from 'react-bootstrap';

import { setDateFormat } from '../../utils/Utils';
import { ACTIVITY_FEED_THEMES } from '../../utils/constants';
import userService from '../../services/user.service';
import stringConstants from '../../utils/stringConstants.json';
import IdfListAvatars from '../idfComponents/idfAdditionalOwners/IdfListAvatars';
import SubmenuDropdownLink from '../commons/DropdownLink';
import activityService from '../../services/activity.service';

const Activity = ({ data, confirm, maxUsers = 4, onHandleEdit }) => {
  const regex =
    // eslint-disable-next-line no-useless-escape
    /^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
  const constants = stringConstants.deals.contacts.profile;
  const [guests, setGuests] = useState([]);

  useEffect(() => {
    const el = document.getElementById(`li-item-date-${data.id}`);
    if (new Date(data?.start_date) < new Date()) {
      el.classList.add('activities-overdue');
    } else if (new Date(data?.start_date).getDay() === new Date().getDay()) {
      el.classList.add('activities-today');
    } else if (new Date(data?.start_date) > new Date()) {
      el.classList.add('activities-planned');
    }
  }, []);

  const markAsDone = async (done) => {
    try {
      await activityService.markAsCompleted(data?.id);
      confirm(constants.updatedActivity);
    } catch (error) {
      confirm(constants.errorUpdatedActivity);
    }
  };

  const getUserByIds = async (guests = '') => {
    if (guests === '') {
      return [];
    }

    const guestIds = guests.split(',');
    const result = [];
    const guestUuid = guestIds.filter((guestId) => {
      if (regex.test(guestId)) {
        result.push({
          value: guestId,
          name: guestId,
          email: guestId,
          avatar: '',
          id: guestId,
          external: true,
        });
      }

      return !regex.test(guestId);
    });

    if (guestUuid.length === 0) {
      return setGuests([...result]);
    }
    const { data: response } = await userService.getGuestsByIds(
      guestUuid.toString()
    );

    const otherData = [];

    response?.users?.forEach((user) => {
      const name = `${user.first_name} ${user.last_name}`;
      otherData.push({
        name,
        value: name,
        first_name: user.first_name,
        last_name: user.last_name,
        title: user.title,
        email: user.email || user.email_work || user.email_home,
        avatar: user.avatar,
        id: user.id,
        url: `/settings/profile/users/${user.id}`,
      });
    });

    response?.contacts?.forEach((item) => {
      const name = `${item.first_name} ${item.last_name}`;
      otherData.push({
        value: name,
        name,
        first_name: item.first_name,
        last_name: item.last_name,
        email: item.email || item.email_work || item.email_home,
        avatar: item.avatar,
        id: item.id,
        url: `/contacts/${item.id}/profile`,
      });
    });

    setGuests([...otherData, ...result]);
  };

  useEffect(() => {
    if (data.guests) {
      getUserByIds(data?.guests);
    }
  }, [data]);

  return (
    <div>
      <div className="card pb-0 border-0">
        <div className="card-body px-2 pt-2 pb-0">
          <div className="text-wrap d-flex flex-row align-items-center justify-content-between">
            <div
              className="custom-control custom-control-green custom-radio"
              title="Mark as done"
            >
              <input
                data-testid={`checkbox-${data.id}`}
                type="checkbox"
                className="custom-control-input"
                id={data.id}
                checked={data?.done}
                onChange={() => {
                  markAsDone(!data.done);
                }}
              />
              <label className="custom-control-label" htmlFor={data.id} />
              <span
                className="text-block cursor-pointer"
                onClick={() => onHandleEdit(data)}
              >
                {data.name}
              </span>
            </div>
            <div className="activity-icon">
              {data?.type ? (
                <span
                  className={`step-icon ${
                    ACTIVITY_FEED_THEMES[data.type].color
                  }`}
                >
                  <i className="material-icons-outlined mr-1">
                    {ACTIVITY_FEED_THEMES[data.type].icon}
                  </i>
                </span>
              ) : null}
            </div>
          </div>
          <div>
            <ul className="list-inline text-muted font-size-xs">
              {data?.start_date && (
                <li
                  className="list-inline-item mr-3 align-middle"
                  id={`li-item-date-${data.id}`}
                  data-testid={`li-item-date-${data.id}`}
                >
                  {setDateFormat(data?.start_date, 'MMM DD YYYY h:mm a')}
                </li>
              )}
              <li className="list-inline-item mr-4 align-middle">
                <Row>
                  <Col className="col-auto px-1">
                    <IdfListAvatars
                      users={guests}
                      maxUsers={maxUsers}
                      defaultSize={'xs'}
                    />
                  </Col>
                  {guests.length > maxUsers ? (
                    <Col className="px-0">
                      <SubmenuDropdownLink
                        classMenu="menu-guest"
                        id={data.id}
                        data={guests}
                        title={
                          <div className="section-owners-header more-owners">
                            <button
                              className={`btn btn-icon btn-xs btn-ghost-primary rounded-circle`}
                            >
                              <i className="material-icons-outlined">add</i>
                            </button>
                          </div>
                        }
                      />
                    </Col>
                  ) : null}
                </Row>
              </li>
              <li className="list-inline-item align-middle">
                {data?.location?.length
                  ? data.location
                  : data?.conference_link?.length
                  ? data.conference_link
                  : null}
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  );
};

export default Activity;
