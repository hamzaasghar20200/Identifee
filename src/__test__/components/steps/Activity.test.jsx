import { fireEvent, render, screen } from '@testing-library/react';
import { waitFor } from '@testing-library/dom';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import moment from 'moment';

import Activity from '../../../components/steps/Activity';
import userService from '../../../services/user.service';
import feedService from '../../../services/feed.service';
import { ACTIVITY_FEED_THEMES } from '../../../utils/constants';
import {
  dataGetGuest,
  dataGetThreeGuest,
} from '../../__mocks__/services/user.service';

jest.mock('../../../services/user.service');
jest.mock('../../../services/feed.service');

describe('Test Actitivy', () => {
  const history = createMemoryHistory();

  const props = {
    data: {
      id: 'data_id_1',
      start_date: '2022-03-09T13:15:00.000Z',
      feed: {
        id: 'feed_id_1',
        summary: 'Activity added',
        type: 'meetingActivity',
      },
      guests: 'guest_id_1',
      done: false,
      name: 'test meet',
      location: 'Meet',
      conference_link: '',
      type: 'callActivity',
    },
  };

  beforeEach(async () => {
    jest.clearAllMocks();
    jest.resetModules();
    jest.restoreAllMocks();
    jest.resetAllMocks();

    userService.getGuestsByIds.mockImplementation(() => Promise.resolve([]));
    feedService.updateActivity.mockImplementation(() => Promise.resolve({}));
  });

  it('Snapshot component', async () => {
    const component = render(<Activity {...props} />);
    await waitFor(() => {
      expect(component.container).toBeVisible();
      expect(component.baseElement).toMatchSnapshot();
    });
  });

  it('Should render icon call', async () => {
    props.data.type = 'callActivity';
    const component = render(<Activity {...props} />);
    const icon = component.getByText('call');
    await waitFor(() => {
      expect(icon).toEqual(
        component.getByText(ACTIVITY_FEED_THEMES[props.data.type].icon)
      );
    });
  });

  it('Should render icon meeting', async () => {
    props.data.type = 'meetingActivity';
    const component = render(<Activity {...props} />);
    const icon = component.getByText('groups');
    await waitFor(() => {
      expect(icon).toEqual(
        component.getByText(ACTIVITY_FEED_THEMES[props.data.type].icon)
      );
    });
  });

  it('Should render icon task', async () => {
    props.data.type = 'taskActivity';
    const component = render(<Activity {...props} />);
    const icon = component.getByText('schedule');
    await waitFor(() => {
      expect(icon).toEqual(
        component.getByText(ACTIVITY_FEED_THEMES[props.data.type].icon)
      );
    });
  });

  it('Should render icon deadline', async () => {
    props.data.type = 'deadlineActivity';
    const component = render(<Activity {...props} />);
    const icon = component.getByText('flag');
    await waitFor(() => {
      expect(icon).toEqual(
        component.getByText(ACTIVITY_FEED_THEMES[props.data.type].icon)
      );
    });
  });

  it('Should render icon email', async () => {
    props.data.type = 'emailActivity';
    const component = render(<Activity {...props} />);
    const icon = component.getByText('email');
    await waitFor(() => {
      expect(icon).toEqual(
        component.getByText(ACTIVITY_FEED_THEMES[props.data.type].icon)
      );
    });
  });

  it('Should render icon lunch', async () => {
    props.data.type = 'lunchActivity';
    const component = render(<Activity {...props} />);
    const icon = component.getByText('restaurant_menu');
    await waitFor(() => {
      expect(icon).toEqual(
        component.getByText(ACTIVITY_FEED_THEMES[props.data.type].icon)
      );
    });
  });

  it('Should render with the format correct', async () => {
    const format = 'MMM DD YYYY h:mm A';
    const component = render(<Activity {...props} />);
    const date = component.getByTestId(`li-item-date-${props.data.id}`);

    await waitFor(() => {
      expect(date).toEqual(
        component.getByText(moment(props.data.start_date).utc().format(format))
      );
    });
  });

  it('Should show the conference link if the data dont have location', async () => {
    props.data.location = '';
    props.data.conference_link = 'conference link';
    const component = render(<Activity {...props} />);

    await waitFor(() => {
      expect(component.baseElement).toHaveTextContent(
        props.data.conference_link
      );
    });
  });

  it('Should call prop confirm at least once', async () => {
    const mockConfirm = jest.fn();
    const component = render(<Activity {...props} confirm={mockConfirm} />);
    const input = component.getByTestId(`checkbox-${props.data.id}`);
    fireEvent.click(input);

    await waitFor(() => {
      expect(mockConfirm).toHaveBeenCalledTimes(1);
    });
  });

  it('Snapshot component with guest rendered', async () => {
    userService.getGuestsByIds.mockImplementation(() =>
      Promise.resolve(dataGetGuest)
    );

    const component = render(
      <Router history={history}>
        <Activity {...props} />
      </Router>
    );
    const initials = `${dataGetGuest.data.users[0].first_name[0]}${dataGetGuest.data.users[0].last_name[0]}`;
    await screen.findByText(initials);

    await waitFor(() => {
      expect(component.baseElement).toHaveTextContent(initials);
      expect(component.baseElement).toMatchSnapshot();
    });
  });

  it('Snapshot component with guests and icon add', async () => {
    userService.getGuestsByIds.mockImplementation(() =>
      Promise.resolve(dataGetThreeGuest)
    );

    const component = render(
      <Router history={history}>
        <Activity {...props} maxUsers={2} />
      </Router>
    );
    const initials = `${dataGetThreeGuest.data.users[0].first_name[0]}${dataGetThreeGuest.data.users[0].last_name[0]}`;
    await screen.findByText(initials);
    const iconAdd = component.getByText('add');

    await waitFor(() => {
      expect(iconAdd).toBeVisible();
      expect(component.baseElement).toMatchSnapshot();
    });
  });
});
