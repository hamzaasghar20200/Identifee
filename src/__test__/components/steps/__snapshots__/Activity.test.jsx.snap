// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Actitivy Snapshot component 1`] = `
<body>
  <div>
    <div>
      <div
        class="card"
      >
        <div
          class="card-body d-flex"
        >
          <div
            class="info mr-2"
          >
            <h4
              class="text-wrap d-flex flex-row align-items-center"
            >
              <div
                class="custom-control custom-control-green custom-radio"
                title="Mark as done"
              >
                <input
                  class="custom-control-input"
                  data-testid="checkbox-data_id_1"
                  id="data_id_1"
                  type="checkbox"
                />
                <label
                  class="custom-control-label"
                  for="data_id_1"
                />
              </div>
              <a
                class="text-block"
                href="#"
              >
                test meet
              </a>
            </h4>
            <ul
              class="list-inline text-muted font-size-xs"
            >
              <li
                class="list-inline-item mr-3 align-middle activities-overdue"
                data-testid="li-item-date-data_id_1"
                id="li-item-date-data_id_1"
              >
                Mar 09 2022 1:15 PM
              </li>
              <li
                class="list-inline-item mr-4 align-middle"
              >
                <div
                  class="row"
                >
                  <div
                    class="col-auto px-1 col"
                  >
                    <div
                      class="align-items-end more-owners"
                    />
                  </div>
                </div>
              </li>
              <li
                class="list-inline-item align-middle"
              >
                Meet
              </li>
            </ul>
          </div>
          <div
            class="activity-icon"
          >
            <span
              class="step-icon bg-soft-green"
            >
              <i
                class="material-icons-outlined"
              >
                call
              </i>
            </span>
          </div>
        </div>
      </div>
      <hr />
    </div>
  </div>
</body>
`;

exports[`Test Actitivy Snapshot component with guest rendered 1`] = `
<body>
  <div>
    <div>
      <div
        class="card"
      >
        <div
          class="card-body d-flex"
        >
          <div
            class="info mr-2"
          >
            <h4
              class="text-wrap d-flex flex-row align-items-center"
            >
              <div
                class="custom-control custom-control-green custom-radio"
                title="Mark as done"
              >
                <input
                  class="custom-control-input"
                  data-testid="checkbox-data_id_1"
                  id="data_id_1"
                  type="checkbox"
                />
                <label
                  class="custom-control-label"
                  for="data_id_1"
                />
              </div>
              <a
                class="text-block"
                href="#"
              >
                test meet
              </a>
            </h4>
            <ul
              class="list-inline text-muted font-size-xs"
            >
              <li
                class="list-inline-item mr-3 align-middle activities-overdue"
                data-testid="li-item-date-data_id_1"
                id="li-item-date-data_id_1"
              >
                Mar 09 2022 1:15 PM
              </li>
              <li
                class="list-inline-item mr-4 align-middle"
              >
                <div
                  class="row"
                >
                  <div
                    class="col-auto px-1 col"
                  >
                    <div
                      class="align-items-end more-owners"
                    >
                      <a
                        class="d-flex"
                        href="/settings/profile/users/guest_1"
                      >
                        <div
                          class="avatar avatar-xs avatar-circle avatar-soft-primary avatar-md mr-n2"
                        >
                          <span
                            class="avatar-initials avatar-icon-font-size"
                          >
                            G1
                          </span>
                        </div>
                      </a>
                    </div>
                  </div>
                </div>
              </li>
              <li
                class="list-inline-item align-middle"
              >
                conference link
              </li>
            </ul>
          </div>
          <div
            class="activity-icon"
          >
            <span
              class="step-icon bg-soft-pink"
            >
              <i
                class="material-icons-outlined"
              >
                restaurant_menu
              </i>
            </span>
          </div>
        </div>
      </div>
      <hr />
    </div>
  </div>
</body>
`;

exports[`Test Actitivy Snapshot component with guests and icon add 1`] = `
<body>
  <div>
    <div>
      <div
        class="card"
      >
        <div
          class="card-body d-flex"
        >
          <div
            class="info mr-2"
          >
            <h4
              class="text-wrap d-flex flex-row align-items-center"
            >
              <div
                class="custom-control custom-control-green custom-radio"
                title="Mark as done"
              >
                <input
                  class="custom-control-input"
                  data-testid="checkbox-data_id_1"
                  id="data_id_1"
                  type="checkbox"
                />
                <label
                  class="custom-control-label"
                  for="data_id_1"
                />
              </div>
              <a
                class="text-block"
                href="#"
              >
                test meet
              </a>
            </h4>
            <ul
              class="list-inline text-muted font-size-xs"
            >
              <li
                class="list-inline-item mr-3 align-middle activities-overdue"
                data-testid="li-item-date-data_id_1"
                id="li-item-date-data_id_1"
              >
                Mar 09 2022 1:15 PM
              </li>
              <li
                class="list-inline-item mr-4 align-middle"
              >
                <div
                  class="row"
                >
                  <div
                    class="col-auto px-1 col"
                  >
                    <div
                      class="align-items-end more-owners"
                    >
                      <a
                        class="d-flex"
                        href="/settings/profile/users/guest_1"
                      >
                        <div
                          class="avatar avatar-xs avatar-circle avatar-soft-primary avatar-md mr-n2"
                        >
                          <span
                            class="avatar-initials avatar-icon-font-size"
                          >
                            G1
                          </span>
                        </div>
                      </a>
                      <a
                        class="d-flex"
                        href="/settings/profile/users/guest_2"
                      >
                        <div
                          class="avatar avatar-xs avatar-circle avatar-soft-primary avatar-md mr-n2"
                        >
                          <span
                            class="avatar-initials avatar-icon-font-size"
                          >
                            G2
                          </span>
                        </div>
                      </a>
                    </div>
                  </div>
                  <div
                    class="px-0 col"
                  >
                    <div
                      class="dropdown"
                      id="dropdown-link-data_id_1"
                    >
                      <div
                        class="cursor-pointer"
                        href=""
                      >
                        <span
                          class=""
                        >
                          <div
                            class="section-owners-header more-owners"
                          >
                            <button
                              class="btn btn-icon btn-xs btn-ghost-primary rounded-circle"
                            >
                              <i
                                class="material-icons-outlined"
                              >
                                add
                              </i>
                            </button>
                          </div>
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </li>
              <li
                class="list-inline-item align-middle"
              >
                conference link
              </li>
            </ul>
          </div>
          <div
            class="activity-icon"
          >
            <span
              class="step-icon bg-soft-pink"
            >
              <i
                class="material-icons-outlined"
              >
                restaurant_menu
              </i>
            </span>
          </div>
        </div>
      </div>
      <hr />
    </div>
  </div>
</body>
`;
